task/sast-snyk-check/0.3/sast-snyk-check.yaml

```diff
@@ -0,0 +1,139 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: sast-snyk-check
+spec:
+  description: >-
+    Scans source code for security vulnerabilities, including common issues such as SQL injection, cross-site scripting (XSS), and code injection attacks using Snyk Code, a Static Application Security Testing (SAST) tool.
+  results:
+    - description: Tekton task test output.
+      name: TEST_OUTPUT
+  params:
+    - name: SNYK_SECRET
+      description: Name of secret which contains Snyk token.
+      default: snyk-secret
+    - name: ARGS
+      type: string
+      description: Append arguments.
+      default: ""
+    - description: Image URL.
+      name: image-url
+      type: string
+      # In a future 0.2 version of the task, drop the default to make this required
+      default: ""
+    - description: Image digest to report findings for.
+      name: image-digest
+      type: string
+      # In a future 0.2 version of the task, drop the default to make this required
+      default: ""
+    - name: SEVERITY_THRESHOLD
+      type: string
+      description: Report only vulnerabilities at the specified level or higher (Options are low, medium or high).
+      default: "high"
+  volumes:
+    - name: snyk-secret
+      secret:
+        secretName: $(params.SNYK_SECRET)
+        optional: true
+  steps:
+    - name: sast-snyk-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.5@sha256:801a105ba0f9c7f58f5ba5cde1a3b4404009fbebb1028779ca2c5de211e94940
```

Comment:
@jperezdealgaba I think we need to update the reference to the image that includes `csdiff`?

Reply:
You a right. As mentioned in the description, until this MR is not merged: https://github.com/konflux-ci/konflux-test/pull/292 we won't be able to do it.

Reply:
FYI: The image has been updated to [version 1.4.6](https://github.com/konflux-ci/konflux-test/releases/tag/v1.4.6). This version contains `csdiff` and `git` packages

Reply:
Unfortunately , the updated container image contains `csdiff-3.4.1-1.el9`, which does not include `csfilter-kfp` yet.

Reply:
How can that be solved? Is the container image needed to be rebuilt again? Do we need to wait until the 3.5.0 version  package is marked as `stable`: https://packages.fedoraproject.org/pkgs/csdiff/csdiff/?

Reply:
Yes, rebuilding the container image after the package appears on mirrors of stable Fedora repositories sounds like the most straightforward approach.  I believe it will be submitted to stable later today as I have asked a few people to test and provide karma at: https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2024-b788a2df5c

Reply:
The package already appears as "stable". I have asked the Konflux team to rebuild the image. Waiting for response

Reply:
Regarding this. Engineering team just told me:

> Hi, we can't rebuild the release, but we can adopt the changes when they get propagated to the konflux-test image.

Reply:
If building an image with updated `csdiff` is too complicated, we can updated `csdiff` in the CI task as an interim solution.  The package update command can be removed later on when the image gets updated.

Reply:
Just got:
> I think we should be able to cut a new release tomorrow

So I will wait until the end of this week and if not, I will add the interim soultion

Reply:
The new container image was released (V1.4.7) (https://quay.io/repository/redhat-appstudio/konflux-test?tab=tags&tag=latest)! I tested and it is containing `csfilter-kfp` : https://konflux.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/application-pipeline/workspaces/jperezde/applications/test-coverity/pipelineruns/osh-cli-container-konflux-test-2-on-pull-request-8wx2v